cadc-download-manager-server/src/main/java/ca/nrc/cadc/dlm/server/ServerUtil.java

```diff
@@ -153,28 +155,177 @@ public static String getCodebase(HttpServletRequest request) {
+     * @param request
+     */
+    public static void handleDeprecatedParams(HttpServletRequest request) {
+        // Check to see if URIs have already been parsed from the request parameters.
+        // If so, this is an internal dispatch/forward being processed
+        List<URI> uriList = (List<URI>)request.getAttribute("uriList");
+        if (uriList == null) {
+            try {
+                String referer = request. getHeader("referer");
+                String[] sa;
+
+                // fileClass -> dynamic params = AD scheme-specific part
+                String[] fileClasses = request.getParameterValues("fileClass");
+                if (fileClasses != null) {
+                    // fileClass is a list of parameters giving other URIs
+                    log.debug("fileClass param(s): " + fileClasses.length);
+                    log.warn("deprecated param 'fileClass' used by " + referer);
+                    for (String fileClass : fileClasses) {
+                        log.debug("fileClass: " + fileClass);
+                        sa = request.getParameterValues(fileClass);
+                        if (sa != null) {
+                            for (String curSa : sa) {
+                                String u = processURI(curSa);
+                                if (u != null) {
+                                    u = toAd(u);
+                                    log.debug("\turi: " + u);
+                                    uriList.add(new URI(u));
+                                }
+                            }
+                        }
+                    }
+                }
+
+                sa = request.getParameterValues("fileId");
+                if (sa != null) {
+                    log.debug("fileId param(s): " + sa.length);
+                    log.warn("deprecated param 'fileId' used by " + referer);
+                    for (String curSa : sa) {
+                        String u = processURI(curSa);
+                        if (u != null) {
+                            u = toAd(u);
+                            uriList.add(new URI(u));
+                        }
+                    }
+                }
+
+                if (!uriList.isEmpty()) {
+                    request.setAttribute("uriList", uriList);
+                }
+            } catch (URISyntaxException ure) {
+                log.error("error parsing URI from deprecated input parameter");
+            }
+        }
+
+        String params = (String) request.getAttribute("params");
+        if (params == null) {
+            Map<String,List<String>> paramMap = ServerUtil.getParameters(request);
+
+            // things to strip out
+            paramMap.remove("fileId");
+            List<String> fcs = paramMap.get("fileClass");
+            if (fcs != null) {
+                for (String fc : fcs) {
+                    paramMap.remove(fc);
+                }
+                paramMap.remove("fileClass");
+            }
+
```

Comment:
I am pretty sure fragment is no longer used by anything. It was used to specify a fragment to attach to each URI before calling the SchemeHandler to convert it to a URL... it was originally used to pass the options like cutout, runid, etc into the SchemeHandler. 

It has never been documented so the chance that someone discovered it and figured out how to use it is very low.

Reply:
Fragment is currently used by Advanced Search to pass in the runid.